cesko-digital/loono
# feat(ci): firebase test lab - backend tests

Vicemene jsem už popisoval/říkal vše, samotný BE test byl přidán v tomto PR: https://github.com/cesko-digital/loono/pull/208

**Fungování ve zkratce:** Vygeneruje se [custom_token](https://firebase.google.com/docs/auth/admin/create-custom-tokens) přes Firebase Admin SDK (python skript). Tento token se vloží do .env, a poté se appka zkompiluje. Během testu se v registrační obrazovce se místo Google/Apple loginu použije přihlašovací metoda [signInWithCustomToken](https://pub.dev/documentation/firebase_auth/latest/firebase_auth/FirebaseAuth/signInWithCustomToken.html), token se načte z .env z assetů. Tímto se vytvoří účet a autentifikuej uživatel aby mohl volat API.

- dá se spustit manuálně + cron - 1x denně? dal jsem v 06:00 UTC
- když failne, notifikace na Slack? ✅

Testovací flow toho 1 BE testu:

```
1. Proceed through the onboarding form
2. Perform sign up with Firebase and send onboarding data via Onboarding API
3. Verify account is created and receives badge(s) and gets valid list of check-ups
4. Verify the user can successfully perform a self-examination and gets rewards from it
5. Verify account details can be changed in the Settings
```

přicemž se otestují tyto API endpointy:

- POST /account/onboard
- UPDATE /account
- GET /examinations
- GET /leaderboard
- POST /examinations/{self-type}/self

Zatím jen 1 BE test, ale je to setupnuté, takže se dá pak jednoduše rozšířit na více BE testů. Jen se pak musí případně upravit py_script aby se generovalo více custom_tokenů (aby se mohlo vytvořit více účtů)
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
